packages/babel-preset-expo/src/use-dom-directive-plugin.ts

```diff
@@ -96,6 +96,16 @@ export default React.forwardRef((props, ref) => {
 });`
         );
 
+        // Removes all imports using babel API, that will disconnect import bindings from the program.
+        // plugin-transform-typescript TSX uses the bindings to remove type imports.
+        // If the DOM component has `import React from 'react';`,
+        // the plugin-transform-typescript treats it as an typed import and removes it.
+        // That will futher cause undefined `React` error.
+        path.traverse({
+          ImportDeclaration(path) {
+            path.remove();
+          },
+        });
```

Comment:
This could be a bit leaky. Maybe we could just inline the `React` usages like `require('react').forwardRef`